[waspls] fix client directory module resolution

craigmc08 opened the pull request:
### Description

Previously, waspls incorrectly thought that importing `@client/file.jsx` could be referencing the file `src/client/file.tsx`. But this JS/TS "merging" is only correct on the server (for reasons that are not explained by the TS documentation on module resolution). This PR fixes that by requiring exact matches for external imports beginning with `@client`.

This incorrect behavior manifested as several different bugs:
- Not reporting missing `.jsx` files when a `.tsx` file with the same name exists.
- Go-to definition going to the `.tsx` file instead of the `.jsx` file when the external import ends in `.jsx`.
- Scaffold actions suggestions scaffolding in both the `.jsx` and `.tsx`files.

This PR should fix all of those bugs.

### Select what type of change this PR introduces:

1. [ ] **Just code/docs improvement** (no functional change). 
2. [X] **Bug fix** (non-breaking change which fixes an issue).
3. [ ] **New feature** (non-breaking change which adds functionality).
4. [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

### Update Waspc ChangeLog and version if needed
If you did a bug fix, new feature, or breaking change, that affects waspc, make sure you satisfy the following:
1. [ ] I updated [ChangeLog.md](https://github.com/wasp-lang/wasp/blob/main/waspc/ChangeLog.md) with description of the change this PR introduces.
2. [ ] I bumped waspc version in [waspc.cabal](https://github.com/wasp-lang/wasp/blob/main/waspc/waspc.cabal) to reflect changes I introduced, with regards to the version of the latest wasp release, if the bump was needed.

infomiho commented on waspc/waspls/src/Wasp/LSP/ExtImport/Path.hs:
```diff
@@ -56,11 +57,18 @@ instance Hashable ExtFileCachePath where
+    Just relPath ->
+      Just $ case SP.splitRelExtension relPath of
+        Nothing -> ExtFileCachePath relPath DotAnyTS
+        Just (withoutExt, ext) ->
```
```suggestion
        Just (pathWithoutExt, ext) ->
```

infomiho commented:
**Ideally:** we would inspect the file's AST and do the changes there and put everything back. It would put imports on top. And, if an import from a file exists, we would just add a new import symbol to the list. This feels like a future improvement.

>  So my thinking was: make it obvious that the import needs to be rewritten a bit by putting it with the rest of the code.

I feel that's a very valid reason to keep it as-is.

craigmc08 commented:
> **Ideally**: we would inspect the file's AST and do the changes there and put everything back. It would put imports on top. And, if an import from a file exists, we would just add a new import symbol to the list. This feels like a future improvement.

I agree, this would definitely be the best way to do it, but will be a lot of work. And it sounds like you're okay with keeping the import in this slightly weird place, so let's stick what that.

I also addressed the comment you left.

infomiho reviewed (approved):
⛴️ it
